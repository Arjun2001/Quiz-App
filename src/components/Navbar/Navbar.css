form{
    background-color: transparent !important;
}

.navbar-bg {
    background-color: #310b0b !important;
}