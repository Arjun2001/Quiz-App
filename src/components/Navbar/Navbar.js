import React from "react";
import { Navbar, Nav, NavDropdown } from "react-bootstrap";
import './Navbar.css'
import {useHistory} from "react-router-dom"
import 'bootstrap/dist/css/bootstrap.css';

function Navigationbar() {
  const history = useHistory();

   const  name= () => {
    let text = 'Signed in as:' + <strong style={{color:"white"}}>{localStorage.getItem('Roll_no')}</strong>;
    return text;
  }

  const logout = () => {
    localStorage.clear();
    history.push('/');
  }


  return (
        <>
        <Navbar collapseOnSelect className="navbar-bg" variant="dark">
        <Navbar.Brand href="/home"><img src="https://cdn.pixabay.com/photo/2018/08/31/11/17/quiz-3644414_960_720.png" style={{width:"60px"}}></img></Navbar.Brand>
        <Navbar.Toggle aria-controls="responsive-navbar-nav" />
        <Navbar.Collapse id="responsive-navbar-nav">
          <Nav className="mr-auto">
          </Nav>  
          <Nav>
            <Navbar.Text>
              Signed in as: <strong style={{color:"white"}}>{localStorage.getItem('Roll_no')}</strong>
            </Navbar.Text>
            <NavDropdown alignRight title={name} id="basic-nav-dropdown">
              <NavDropdown.Item href="/profile">Profile</NavDropdown.Item>
              <NavDropdown.Item onClick={logout}>Logout</NavDropdown.Item>
            </NavDropdown>
          </Nav>
        </Navbar.Collapse>
      </Navbar>
    </>
  );
};

export default Navigationbar;