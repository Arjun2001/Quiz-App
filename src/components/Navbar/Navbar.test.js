import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import Navbar from './Navbar';

describe('Navbar component', () => {
    it('starts with a count of 0', () => {
    const wrapper = shallow(<Navbar />);
    var text = "Signed in as:"
    expect(wrapper.contains(text));
    // const text = wrapper.find('p').text();
    // expect(text).toEqual('Count: 0');
    });


    it('matches the snapshot', () => {
        const tree = renderer.create(<Navbar />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});