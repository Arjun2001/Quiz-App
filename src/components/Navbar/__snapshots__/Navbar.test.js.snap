// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component matches the snapshot 1`] = `
<nav
  className="navbar-bg navbar navbar-expand navbar-dark"
>
  <a
    className="navbar-brand"
    href="/home"
  >
    <img
      src="https://cdn.pixabay.com/photo/2018/08/31/11/17/quiz-3644414_960_720.png"
      style={
        Object {
          "width": "60px",
        }
      }
    />
  </a>
  <button
    aria-controls="responsive-navbar-nav"
    aria-label="Toggle navigation"
    className="navbar-toggler collapsed"
    onClick={[Function]}
    type="button"
  >
    <span
      className="navbar-toggler-icon"
    />
  </button>
  <div
    aria-expanded={null}
    className="navbar-collapse collapse"
    id="responsive-navbar-nav"
  >
    <div
      className="mr-auto navbar-nav"
      onKeyDown={[Function]}
    />
    <div
      className="navbar-nav"
      onKeyDown={[Function]}
    >
      <span
        className="navbar-text"
      >
        Signed in as: 
        <strong
          style={
            Object {
              "color": "white",
            }
          }
        />
      </span>
      <div
        className="nav-item dropdown"
      >
        <a
          aria-expanded={false}
          aria-haspopup={true}
          className="dropdown-toggle nav-link"
          data-rb-event-key={null}
          href="#"
          id="basic-nav-dropdown"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
        />
      </div>
    </div>
  </div>
</nav>
`;
